MULTIPLATFORM.md

```diff
@@ -0,0 +1,244 @@
+`common` project. Let's implement our `expect`-class:
+
+    // greeting/android/src/main/kotlin/android.kt
+    package org.greeting
+
+    actual class Platform actual constructor() {
+        actual val platform: String = "Android"
+    }
+
+We build this project into a Java-library which can be used by Android Studio project as a dependency. So the content
+of `android/build.gradle` will be the following:
+
+    apply plugin: 'java-library'
+    apply plugin: 'kotlin-platform-jvm'
+
+    // Specify a group and a version of the library to access it in Android Studio.
+    group = 'org.greeting'
+    version = 1.0
+
+    dependencies {
+        // Specify Kotlin/JVM stdlib dependency.
+        implementation "org.jetbrains.kotlin:kotlin-stdlib-jre7:$kotlin_version"
+
+        // Specify dependency on a common project for Kotlin multiplatform build.
+        expectedBy project(':common')
+    }
+
+#### 2.3 iOS subproject
+
+As well as `android`, this project contains platform-dependent implementations of `expect`-declarations:
+
+    // greeting/ios/src/main/kotlin/ios.kt
+    package org.greeting
+
+    actual class Platform actual constructor() {
+        actual val platform: String = "iOS"
+    }
+
+We build this project into an Objective-C framework using Kotlin/Native compiler. To do this, declare a framework in
+`ios/build.gradle` and add an `expectedBy` dependency in the same manner as in the Android project:
+
+    apply plugin: 'konan'
+
+    konanArtifacts {
+        // Declare building into a framework.
+        framework('ios-greeting') {
+            // The multiplatform support is disabled by default.
+            enableMultiplatform true
+        }
+    }
+
+    dependencies {
+        // Specify dependency on a common project for Kotlin multiplatform build
+        expectedBy project(':common')
+    }
+
+### 3. Android application
+
+Now we can create an Android application which will use the library we implemented on the previous step. Open Android
+Studio and create a new project in the `androidApp` directory. Android Studio generates all necessary files and
+directories so we need only add a dependency on our library. There are only 2 actions we have to do:
+
+1. Add dependency on the library. To do this just open `androidApp/app/build.gradle` and add the following snippet in
+the `dependencies` script block:
+
+    ```
+    implementation 'org.greeting:android:1.0'
+    implementation 'org.greeting:common:1.0'
+    ```
+2. Include `greeting` build in the Android Studio project as part of
+[composite build](https://docs.gradle.org/current/userguide/composite_builds.html). To do it, add the
+following line in `androidApp/settings.gradle`:
+
+    ```
+    includeBuild '../greeting'
+    ```
+    Now dependencies of the application can be resolved in artefacts built by `greeting`, so we can access our library.
+
+**TODO: What is better: composite build or adding `greeting` as a subproject in the AS project?**
```

Comment:
Well. Great question 👍 

From my point of view it depends on the project.
If we assume that the "shared code" (`greeting/`) is developed by another team than the "android code" (`android/`) I would do it with composite builds (in real life it would be provided by some maven repo).
If we assume that the shared code is developed by the **same** team than the android code (and the ios code) than I would include it as a subproject. Why? Because you don't have to change your IDE to make changes in the shared code. You do everything directly in AS.

For that sample I would use a includeBuild but would **definitive** document/note it that it is possible via a subproject 👍

Reply:
Agree, It worth describing both these options. Will add it.

Reply:
Hm, I just opened the project with composite build in Android Studio (3.1) and it correctly works with sources of the included build. Also will not adding `greeting` in the `androidApp` as a subproject look hacky because in this case these projects will be included into 2 Gradle builds (the root one and the `androidApp` build)?